### PR TITLE
Add Ubuntu 22.04 release package

### PR DESCRIPTION
**Pull request by thomasspriggs**
This PR adds a Ubuntu 22.04 a release package job.

Initially raised as a draft PR as the job is currently untested. The Ubuntu 22.04 binaries should already be tested as part of the Pull Request jobs.

A Ubuntu 22.04 package was requested here - https://github.com/diffblue/cbmc/issues/7474

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: .github/workflows/release-packages.yaml [hunk 1]
```diff
@@ -6,6 +6,81 @@ env:
...
+            ${{ runner.os }}-22.04-Release
+      - name: ccache environment
+        run: |
+          echo "CCACHE_BASEDIR=$PWD" >> $GITHUB_ENV
+          echo "CCACHE_DIR=$PWD/.ccache" >> $GITHUB_ENV
+      - name: Configure CMake
+        run: |
+          mkdir build
+          cd build
+          cmake .. -G Ninja -DCMAKE_BUILD_TYPE=Release
```
Comment (tautschnig) [inline .github/workflows/release-packages.yaml]: Can you please use exactly the same `cmake` command line (`cmake -S . -Bbuild -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++`) as the existing job using a `${{ runner.os }}-22.04-Release` prefix for ccache? Perhaps we'd end up with the same build commands then all is well, but I want to make absolutely sure we don't end up trashing caches by using conflicting build command lines.

File: .github/workflows/release-packages.yaml [hunk 2]
```diff
@@ -6,6 +6,81 @@ env:
...
+      - name: Zero ccache stats and limit in size
+        run: ccache -z --max-size=500M
+      - name: Build using Ninja
+        run: |
+          cd build
+          ninja -j2
```
Comment (tautschnig) [inline .github/workflows/release-packages.yaml]: Nit pick: we could save on vertical space by using `ninja -C build -j2`

Comment (thomasspriggs) [reply .github/workflows/release-packages.yaml]: Updated. The job I added matched the existing packaged release jobs, so I updated the other Ubuntu package jobs as well.

File: .github/workflows/release-packages.yaml (hunk 1 above)
Comment (thomasspriggs) [reply .github/workflows/release-packages.yaml]: This should now be fixed.

**Comment by thomasspriggs** [comment]
What we have here seems reasonable to me. I propose that we merge this and see if it works next time we do a release. @NlightNFotis ?

**Review by NlightNFotis — APPROVED** [review]

**Review by TGWDB — APPROVED** [review]

**Pull request merged**